engine/apps/alerts/models/resolution_note.py

```diff
@@ -75,17 +83,17 @@ class ResolutionNoteSlackMessage(models.Model):
     class Meta:
         unique_together = ("thread_ts", "ts")
 
-    def get_resolution_note(self):
+    def get_resolution_note(self) -> typing.Optional["ResolutionNote"]:
         try:
             return self.resolution_note
         except ResolutionNoteSlackMessage.resolution_note.RelatedObjectDoesNotExist:
             return None
 
-    def delete(self):
+    def delete(self, *args, **kwargs) -> typing.Tuple[int, typing.Dict[str, int]]:
         resolution_note = self.get_resolution_note()
         if resolution_note:
             resolution_note.delete()
-        super().delete()
+        return super().delete(*args, **kwargs)
```

Comment:
`mypy` caught a legitimate error here which was that you cannot override an inherited method if it does not have the same function signature. `models.Model.delete` takes several optional args/kwargs